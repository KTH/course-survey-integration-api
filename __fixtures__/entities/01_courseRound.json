{
  "_id": "",
  "id": "",
  "ladokCourseId": "",
  "ladokCourseRoundId": "",
  "canvasSisId": "",
  "name": "",
  "courseCode": "",
  "language": "en",
  "canceled": false,
  "endDate": "",
  "displayYear": "",
  "organization": null,
  "institution": null,
  "courseGoal": "",
  "period": "P1",
  "credits": "",
  "courseExaminor": null,
  "courseResponsible": null,
  "courseTeachers": [],
  "nrofRegisteredStudents": 20,
  "nrofReportedResults": 20,
  "gradingDistribution": {
    "P": 1
  },
  "programs": [],
  "modules": []
}
