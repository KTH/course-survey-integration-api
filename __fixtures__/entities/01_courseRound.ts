import { TCourseRoundEntity } from "../../src/functions/interface"

export const courseRound1: TCourseRoundEntity = {
  id: "01-1",
  language: "sv",
  canceled: false,
  institution: {
    displayName: "Skolan för elektroteknik och datavetenskap",
    displayCode: "ITM",
    kthId: "u9itm000"
  },
  periods: [{ period: "P1", credits: "7.5 hp" }],
  courseExaminers: [
    { userName: "sven", kthUserId: "u9sven00", email: "sven@example.com", fullName: "Sven Svensson" },
    { userName: "anna", kthUserId: "u9anna00", email: "anna@example.com", fullName: "Anna Annasdotter" }
  ],
  ladokCourseId: "8734afdf-1da9-415d-8c08-9ed160f87c0a-ladokCourseId",
  ladokCourseRoundId: "41374a41-2bda-404f-b11c-6e007deb61b4-ladokCoursRoundId",
  canvasSisId: "41374a41-2bda-404f-b11c-6e007deb61b4-ladokCoursRoundId",
  name: "Introduktion till programmering",
  courseGoal: "Förståelse av grundläggande koncept: Studenten bör kunna förklara och tillämpa grundläggande begrepp inom programmering, såsom variabler, datatyper, villkorssatser, loopar och funktioner.",
  organization: { displayName: "Institution för datavetenskap", displayCode: "ITM/JA", kthId: "u9ja0000" },
  courseResponsible: [
    {
      userName: "emil",
      kthUserId: "u9emil00",
      email: "emil@example.com",
      fullName: "Emil Emilsson"
    }
  ],
  courseTeachers: [
    {
      userName: "elin",
      kthUserId: "u9elin00",
      email: "elin@example.com",
      fullName: "Elin Elinsson"
    }
  ],
  _gradingScheme: ["P", "F"],
  courseCode: "AA0000",
  endDate: "2021-01-01",
  displayYear: "2021",
  credits: "7.5 hp",
  modules: [
    {
      moduleRoundId: "531ea06f-a5dd-40f4-8709-cfd846e6e09d-ladokModuleRoundId",
      code: "TEN1",
      name: "Tentamen",
      credits: "7.5 hp",
      gradingScheme: ["P", "F"]
    }
  ]
}
