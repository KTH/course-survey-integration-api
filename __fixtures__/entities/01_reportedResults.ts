import { TReportedResultEntity } from "../../src/functions/interface";

export const reportedResults1: TReportedResultEntity[] = [
  {
    id: "01-1",
    parentId: "531ea06f-a5dd-40f4-8709-cfd846e6e09d-ladokModuleRoundId",
    ladokCourseRoundId:
      "41374a41-2bda-404f-b11c-6e007deb61b4-ladokCoursRoundId",
    hashedStudentId: "0cd086ab-78ee-4b7c-a785-29821b364111-hashedStudentId",
    decision: "2fb3cbf5-289c-4905-854b-a2c450ee7267-beslutsId",
    result: "P",
    metaData: {
      HandelseUID: "0f77af27-8d9b-4e9c-b2fe-600fc1cec9ad-handelseId",
      BetygsgradID: 131661,
      BetygsskalaID: 131657,
      ResultatUID: "c4d91b0a-a0c8-4695-9f46-b8c93fbb3248-resultatId",
    },
  },
  {
    id: "01-2",
    parentId: "8734afdf-1da9-415d-8c08-9ed160f87c0a-ladokCourseId",
    ladokCourseRoundId:
      "41374a41-2bda-404f-b11c-6e007deb61b4-ladokCoursRoundId",
    hashedStudentId: "0cd086ab-78ee-4b7c-a785-29821b364111-hashedStudentId",
    decision: "207833e4-79d7-459d-8317-ac522bc33c8e-beslutsId",
    result: "P",
    metaData: {
      HandelseUID: "48c98586-fc0b-46fc-b8a7-5c28f5716e84-handelseId",
      BetygsgradID: 131661,
      BetygsskalaID: 131657,
      ResultatUID: "9daefc8b-153c-4910-9e0b-91a014cbf6da-resultatId",
    },
  },
  {
    id: "01-3",
    parentId: "531ea06f-a5dd-40f4-8709-cfd846e6e09d-ladokModuleRoundId",
    ladokCourseRoundId:
      "41374a41-2bda-404f-b11c-6e007deb61b4-ladokCoursRoundId",
    hashedStudentId: "4c18639a-7e9d-4ebc-9f90-92b40cbd5508-hashedStudentId",
    decision: "aa9ec3d9-ea6b-461b-a7d4-d40bfb136a96-beslutsId",
    result: "P",
    metaData: {
      HandelseUID: "20fd6357-786b-4801-8f92-bfa055620c8f-handelseId",
      BetygsgradID: 131661,
      BetygsskalaID: 131657,
      ResultatUID: "0dc2d67a-df1b-41c9-af76-6cc358a16759-resultatId",
    },
  },
  {
    id: "01-4",
    parentId: "8734afdf-1da9-415d-8c08-9ed160f87c0a-ladokCourseId",
    ladokCourseRoundId:
      "41374a41-2bda-404f-b11c-6e007deb61b4-ladokCoursRoundId",
    hashedStudentId: "4ff5969a-0e54-11ed-8f4d-87c157374df8-hashedStudentId",
    decision: "e5d84a88-847a-48e3-a643-88ab415ba90f-beslutsId",
    result: "P",
    metaData: {
      HandelseUID: "f75b0757-1183-4533-8201-b9de39ac264c-handelseId",
      BetygsgradID: 131661,
      BetygsskalaID: 131657,
      ResultatUID: "1e380e11-b44f-11ee-988f-6acd08c746d6",
    },
  },
  {
    id: "01-5r",
    parentId: "531ea06f-a5dd-40f4-8709-cfd846e6e09d-ladokModuleRoundId",
    ladokCourseRoundId:
      "41374a41-2bda-404f-b11c-6e007deb61b4-ladokCoursRoundId",
    hashedStudentId: "1e380e11-b44f-11ee-988f-6acd08c746d6-hashedStudentId",
    decision: "2fb3cbf5-289c-4905-854b-a2c450ee7267-beslutsId",
    result: "P",
    metaData: {
      HandelseUID: "0f77af27-8d9b-4e9c-b2fe-600fc1cec9ad-handelseId",
      BetygsgradID: 131661,
      BetygsskalaID: 131657,
      ResultatUID: "1e380e11-b44f-11ee-988f-6acd08c746d6-resultatId",
    },
    retraction: {
      Anteckning: "Det bidde fel i systemet. Jag har makulerat beslutet.",
      BeslutUID: "34b1ff17-603e-11e9-9dcc-b1e66e1540b0",
      Beslutsdatum: "2024-01-16",
      Beslutsfattare: "Emil Stenberg (IT)",
      BeslutsfattareUID: "34b1ff17-603e-11e9-9dcc-b1e66e1540b0"
    }
  },
  {
    id: "01-6r",
    metaData: {
      ResultatUID: "1e380e11-b44f-11ee-988f-6acd08c746d6-resultatId",
    },
    retraction: {
      Anteckning: "Det bidde fel i systemet. Jag har makulerat beslutet.",
      BeslutUID: "34b1ff17-603e-11e9-9dcc-b1e66e1540b0",
      Beslutsdatum: "2024-01-16",
      Beslutsfattare: "Emil Stenberg (IT)",
      BeslutsfattareUID: "34b1ff17-603e-11e9-9dcc-b1e66e1540b0"
    }
  },
];
