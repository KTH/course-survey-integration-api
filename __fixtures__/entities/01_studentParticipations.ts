import { TStudentParticipationEntity } from "../../src/functions/interface";

export const StudentParticipation2: TStudentParticipationEntity[] = [
  {
    id: "01-1",
    parentId: "41374a41-2bda-404f-b11c-6e007deb61b4-ladokCoursRoundId",
    hashedStudentId: "0cd086ab-78ee-4b7c-a785-29821b364111-hashedStudentId",
    ladokCourseRoundId:
      "41374a41-2bda-404f-b11c-6e007deb61b4-ladokCoursRoundId",
    canvasSisId: "u9sebbe0",
    name: "Sebastian Sebsson",
    email: "sebbe@example.com",
    roles: ["student"],
    program: {
      code: "ARKIT",
      name: "Arkitektur",
      required: "obligatorisk",
      startTerm: "20171",
      studyYear: 4,
    },
  },
  {
    id: "01-2",
    parentId: "41374a41-2bda-404f-b11c-6e007deb61b4-ladokCoursRoundId",
    hashedStudentId: "4c18639a-7e9d-4ebc-9f90-92b40cbd5508-hashedStudentId",
    ladokCourseRoundId:
      "41374a41-2bda-404f-b11c-6e007deb61b4-ladokCoursRoundId",
    canvasSisId: "u9marta0",
    name: "Marta Martsdottir",
    email: "Marta@example.com",
    roles: ["student"],
    program: {
      code: "ARKIT",
      name: "Arkitektur",
      required: "obligatorisk",
      startTerm: "20171",
      studyYear: 4,
    },
  },
];
