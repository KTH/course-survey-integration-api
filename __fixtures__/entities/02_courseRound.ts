import { TCourseRoundEntity } from "../../src/functions/interface";

export const courseRound2: TCourseRoundEntity = {
  id: "02-1",
  language: "sv",
  canceled: false,
  institution: {
    displayName: "Institutionen för datalogi och datorlingvistik",
    displayCode: "CSC/PK",
    kthId: "u9pk0000",
  },
  periods: [{ period: "P3", credits: "6 hp" }],
  courseExaminers: [
    {
      userName: "karin",
      kthUserId: "u9karin0",
      email: "karin@example.com",
      fullName: "Karin Karlsson",
    },
    {
      userName: "gunnar",
      kthUserId: "u9gunnar",
      email: "gunnar@example.com",
      fullName: "Gunnar Gunnarsson",
    },
  ],
  ladokCourseId: "21a5ec0a-ea41-4967-8eb2-a824b5a720af-ladokCourseId",
  ladokCourseRoundId: "d682b090-e019-4fdf-8675-f10233394e12-ladokCoursRoundId",
  canvasSisId: "d682b090-e019-4fdf-8675-f10233394e12-ladokCoursRoundId",
  name: "Kemiteknik och Bioteknik med matematiska metoder I",
  courseGoal:
    "Förvärva en djupare förståelse för kemiska reaktioner, kinetik, termodynamik och andra grundläggande principer inom kemi.",
  organization: {
    displayName: "Skolan för matematik och systemteknik",
    displayCode: "CSC",
    kthId: "u9csc000",
  },
  courseResponsible: [
    {
      userName: "per",
      kthUserId: "u9per000",
      email: "per@example.com",
      fullName: "Per Persson",
    },
  ],
  courseTeachers: [
    {
      userName: "elisabeth",
      kthUserId: "u9elisab",
      email: "elisabeth@example.com",
      fullName: "Elisabeth Eliadotter",
    },
  ],
  _gradingScheme: ["A", "B", "C", "D", "E", "Fx", "F"],
  courseCode: "BB0001",
  courseInstanceCode: "34210",
  courseInstanceArchivingCode: "BB0001 HT23 51210",
  endDate: "2023-09-01",
  displayYear: "2023",
  credits: "6 hp",
  modules: [
    {
      moduleRoundId: "230f8e64-88fb-4913-83e8-023a472d859f-ladokModuleRoundId",
      code: "TEN1",
      canceled: false,
      name: "Tentamen",
      credits: "5 hp",
      gradingScheme: ["A", "B", "C", "D", "E", "Fx", "F"],
    },
    {
      moduleRoundId: "5e6fafb5-56d4-4c6c-9738-a2d3dbed4354-ladokModuleRoundId",
      code: "LAB",
      canceled: false,
      name: "Laboration",
      credits: "1 hp",
      gradingScheme: ["P", "F"],
    },
  ],
};
