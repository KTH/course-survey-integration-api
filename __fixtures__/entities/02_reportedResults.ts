import { TReportedResultEntity } from "../../src/functions/interface";

export const reportedResults1: TReportedResultEntity[] = [
  {
    id: "02-1",
    parentId: "230f8e64-88fb-4913-83e8-023a472d859f-ladokModuleRoundId",
    ladokCourseRoundId:
      "d682b090-e019-4fdf-8675-f10233394e12-ladokCoursRoundId",
    hashedStudentId: "4cf163f8-c2a5-4f32-9eaf-4447341cfe99-hashedStudentId",
    decision: "57dff692-15a7-4943-9a0d-3b2e3220a723-beslutsId",
    result: "B",
    metaData: {
      HandelseUID: "2161479b-a38b-4035-8755-7a406ff9e915-handelseId",
      BetygsgradID: 131662,
      BetygsskalaID: 131659,
      ResultatUID: "27cc1963-f980-4c8a-b74a-b646b9c6d0c6-resultatId",
    },
  },
  {
    id: "02-2",
    parentId: "5e6fafb5-56d4-4c6c-9738-a2d3dbed4354-ladokModuleRoundId",
    ladokCourseRoundId:
      "d682b090-e019-4fdf-8675-f10233394e12-ladokCoursRoundId",
    hashedStudentId: "4cf163f8-c2a5-4f32-9eaf-4447341cfe99-hashedStudentId",
    decision: "0e72569a-57d3-484c-86bb-68a833f527cc-beslutsId",
    result: "P",
    metaData: {
      HandelseUID: "1a34ec97-9243-4b40-abce-55923944be42-handelseId",
      BetygsgradID: 131661,
      BetygsskalaID: 131657,
      ResultatUID: "e1547f25-9688-4445-99a6-20370db1c7d5-resultatId",
    },
  },
  {
    id: "02-3",
    parentId: "21a5ec0a-ea41-4967-8eb2-a824b5a720af-ladokCourseId",
    ladokCourseRoundId:
      "d682b090-e019-4fdf-8675-f10233394e12-ladokCoursRoundId",
    hashedStudentId: "4cf163f8-c2a5-4f32-9eaf-4447341cfe99-hashedStudentId",
    decision: "7cbf5cbe-1081-4e84-8aa4-a80974e786ce-beslutsId",
    result: "A",
    metaData: {
      HandelseUID: "eb599025-6628-4290-8e53-9b195ba6f18f-handelseId",
      BetygsgradID: 131663,
      BetygsskalaID: 131659,
      ResultatUID: "b6da9442-2181-4a1d-90b8-e777a98d2918-resultatId",
    },
  },
  {
    id: "02-4",
    parentId: "230f8e64-88fb-4913-83e8-023a472d859f-ladokModuleRoundId",
    ladokCourseRoundId:
      "d682b090-e019-4fdf-8675-f10233394e12-ladokCoursRoundId",
    hashedStudentId: "ca3dd14e-3e58-464c-91e1-4c21cbd3a46b-hashedStudentId",
    decision: "86ff2394-a626-4d55-8273-894540af29e0-beslutsId",
    result: "F",
    metaData: {
      HandelseUID: "a9d798af-5328-45c7-a569-2a5ef3e1a48a-handelseId",
      BetygsgradID: 131664,
      BetygsskalaID: 131659,
      ResultatUID: "1916087c-cce7-46ee-8f06-a1aca73c5bcc-resultatId",
    },
  },
];
