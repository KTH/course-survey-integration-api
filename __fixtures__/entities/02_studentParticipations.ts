import { TStudentParticipationEntity } from "../../src/functions/interface";

export const StudentParticipation2: TStudentParticipationEntity[] = [
  {
    id: "02-1",
    parentId: "d682b090-e019-4fdf-8675-f10233394e12-ladokCoursRoundId",
    hashedStudentId: "4cf163f8-c2a5-4f32-9eaf-4447341cfe99-hashedStudentId",
    ladokCourseRoundId:
      "d682b090-e019-4fdf-8675-f10233394e12-ladokCoursRoundId",
    canvasSisId: "u9rasmus",
    name: "Rasmus Rasmusson",
    email: "rust@example.com",
    roles: ["student"],
    program: {
      code: "BIO",
      name: "Biologi och bioteknik",
      required: "obligatorisk",
      startTerm: "20231",
      studyYear: 1,
      specialization: {
        code: "BMRU",
        name: "Biokemi och molekyl√§rbiologianalys med Rust",
      },
    },
  },
  {
    id: "02-2",
    parentId: "d682b090-e019-4fdf-8675-f10233394e12-ladokCoursRoundId",
    hashedStudentId: "ca3dd14e-3e58-464c-91e1-4c21cbd3a46b-hashedStudentId",
    ladokCourseRoundId:
      "d682b090-e019-4fdf-8675-f10233394e12-ladokCoursRoundId",
    canvasSisId: "u9sara00",
    name: "Sara Sarasdottir",
    email: "sara@example.com",
    roles: ["student"],
    program: {
      code: "BIO",
      name: "Biologi och bioteknik",
      required: "obligatorisk",
      startTerm: "20231",
      studyYear: 1,
    },
  },
];
