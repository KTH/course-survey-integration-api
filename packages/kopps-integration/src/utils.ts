import { InvalidLadokUidError, SyllabusNotFoundError } from "./errors";
import type {
  KoppsCourseDetailedInformation,
  KoppsCourseRoundInfo,
  KoppsSyllabus,
} from "./types";

const PERIODS = ["P0", "P1", "P2", "P3", "P4", "P5"] as const;
type Periods = (typeof PERIODS)[number];

/**
 * Return the periods in a round object. Periods are sorted from oldest to
 * latest by the term where the period belongs.
 */
export function getPeriods(roundObject: KoppsCourseRoundInfo) {
  const periods: Periods[] = [];
  const sortedTerms = roundObject.round.courseRoundTerms.sort(
    (t1, t2) => t1.term.term - t2.term.term,
  );

  for (const c of sortedTerms) {
    // Add the number to the "periods" array if the number of credits is higher than 0
    for (const p of PERIODS) {
      const numberOfCredits = c[`credits${p}`];
      if (numberOfCredits && numberOfCredits > 0) {
        periods.push(p);
      }
    }
  }

  return periods;
}

/** Return the course round information given a Ladok UID */
export function getCourseRoundInformation(
  detailedInformation: KoppsCourseDetailedInformation,
  ladokUid: string,
): KoppsCourseRoundInfo {
  const roundInformation = detailedInformation.roundInfos.find(
    (r) => r.round.ladokUID === ladokUid,
  )!;

  if (!roundInformation) {
    throw new InvalidLadokUidError(ladokUid);
  }

  return roundInformation;
}

/** Return the most recent syllabus that is valid in the given term */
export function getSyllabus(
  allSyllabus: KoppsSyllabus[],
  term: number,
): KoppsSyllabus {
  // Filter-out the invalid ones
  const validSyllabus = allSyllabus.filter((s) => s.validFromTerm.term <= term);

  if (validSyllabus.length === 0) {
    throw new SyllabusNotFoundError(term);
  }

  // Take the latest
  return validSyllabus.sort(
    (s1, s2) => s2.validFromTerm.term - s1.validFromTerm.term,
  )[0];
}
