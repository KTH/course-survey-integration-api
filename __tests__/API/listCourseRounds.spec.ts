import handler from "../../src/functions/api/listCourseRounds";
import { MockContext } from "../utils/mockContext";
import { MockDatabase } from "../utils/mockDatabase";
import { MockRequest } from "../utils/mockRequest";

// Mock data
import { courseRound1 } from "../../__fixtures__/entities/01_courseRound";
import { courseRound2 } from "../../__fixtures__/entities/02_courseRound";
import { studentParticipations1 } from "../../__fixtures__/entities/01_studentParticipations";
import { studentParticipations2 } from "../../__fixtures__/entities/02_studentParticipations";
import { reportedResults1 } from "../../__fixtures__/entities/01_reportedResults";
import { reportedResults2 } from "../../__fixtures__/entities/02_reportedResults";

const mockDbData = {
  CourseRound: [
    courseRound1,
    courseRound2,
  ],
  ReportedResult: [
    ...reportedResults1,
    ...reportedResults2,

  ],
  StudentParticipation: [
    ...studentParticipations1,
    ...studentParticipations2,
  ],
}

describe("/api/course-rounds", () => {
  test("can be executed", async () => {
    const mockDb = new MockDatabase(mockDbData);
    const mockContext = new MockContext();
    const mockRequest = new MockRequest();
    const outp = await handler(mockRequest, mockContext, mockDb);
    expect(outp).toMatchSnapshot();
  });
});
