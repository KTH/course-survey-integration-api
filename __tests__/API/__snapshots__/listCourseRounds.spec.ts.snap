// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/api/course-rounds can be executed 1`] = `
{
  "headers": {
    "Pagination-Limit": "30",
    "Pagination-Offset": "0",
    "Pagination-Total-Count": "2",
  },
  "jsonBody": [
    {
      "canceled": false,
      "canvasSisId": "41374a41-2bda-404f-b11c-6e007deb61b4-ladokCoursRoundId",
      "courseCode": "AA0000",
      "courseExaminers": [
        {
          "email": "sven@example.com",
          "fullName": "Sven Svensson",
          "kthUserId": "u9sven00",
          "userName": "sven",
        },
        {
          "email": "anna@example.com",
          "fullName": "Anna Annasdotter",
          "kthUserId": "u9anna00",
          "userName": "anna",
        },
      ],
      "courseGoal": "Förståelse av grundläggande koncept: Studenten bör kunna förklara och tillämpa grundläggande begrepp inom programmering, såsom variabler, datatyper, villkorssatser, loopar och funktioner.",
      "courseResponsible": [
        {
          "email": "emil@example.com",
          "fullName": "Emil Emilsson",
          "kthUserId": "u9emil00",
          "userName": "emil",
        },
      ],
      "courseTeachers": [
        {
          "email": "elin@example.com",
          "fullName": "Elin Elinsson",
          "kthUserId": "u9elin00",
          "userName": "elin",
        },
      ],
      "credits": "7.5 hp",
      "displayYear": "2023-2024",
      "endDate": "2024-01-01",
      "id": "01-1",
      "institution": {
        "displayCode": "ITM/JA",
        "displayName": "Institution för datavetenskap",
        "kthId": "u9ja0000",
      },
      "ladokCourseId": "8734afdf-1da9-415d-8c08-9ed160f87c0a-ladokCourseId",
      "ladokCourseRoundId": "41374a41-2bda-404f-b11c-6e007deb61b4-ladokCoursRoundId",
      "language": "sv",
      "name": "Introduktion till programmering",
      "organization": {
        "displayCode": "ITM",
        "displayName": "Skolan för elektroteknik och datavetenskap",
        "kthId": "u9itm000",
      },
      "periods": [
        {
          "credits": "7.5 hp",
          "period": "P1",
        },
      ],
      "programs": [
        {
          "code": "ARKIT",
          "name": "Arkitektur",
          "required": "obligatorisk",
          "startTerm": "20221",
          "studyYear": 4,
        },
      ],
    },
    {
      "canceled": false,
      "canvasSisId": "d682b090-e019-4fdf-8675-f10233394e12-ladokCoursRoundId",
      "courseCode": "BB0001",
      "courseExaminers": [
        {
          "email": "karin@example.com",
          "fullName": "Karin Karlsson",
          "kthUserId": "u9karin0",
          "userName": "karin",
        },
        {
          "email": "gunnar@example.com",
          "fullName": "Gunnar Gunnarsson",
          "kthUserId": "u9gunnar",
          "userName": "gunnar",
        },
      ],
      "courseGoal": "Förvärva en djupare förståelse för kemiska reaktioner, kinetik, termodynamik och andra grundläggande principer inom kemi.",
      "courseResponsible": [
        {
          "email": "per@example.com",
          "fullName": "Per Persson",
          "kthUserId": "u9per000",
          "userName": "per",
        },
      ],
      "courseTeachers": [
        {
          "email": "elisabeth@example.com",
          "fullName": "Elisabeth Eliadotter",
          "kthUserId": "u9elisab",
          "userName": "elisabeth",
        },
      ],
      "credits": "6 hp",
      "displayYear": "2023",
      "endDate": "2023-09-01",
      "id": "02-1",
      "institution": {
        "displayCode": "CSC/PK",
        "displayName": "Institutionen för datalogi och datorlingvistik",
        "kthId": "u9pk0000",
      },
      "ladokCourseId": "21a5ec0a-ea41-4967-8eb2-a824b5a720af-ladokCourseId",
      "ladokCourseRoundId": "d682b090-e019-4fdf-8675-f10233394e12-ladokCoursRoundId",
      "language": "sv",
      "name": "Kemiteknik och Bioteknik med matematiska metoder I",
      "organization": {
        "displayCode": "CSC",
        "displayName": "Skolan för matematik och systemteknik",
        "kthId": "u9csc000",
      },
      "periods": [
        {
          "credits": "6 hp",
          "period": "P3",
        },
      ],
      "programs": [
        {
          "code": "BIO",
          "name": "Biologi och bioteknik",
          "required": "obligatorisk",
          "specialization": {
            "code": "BMRU",
            "name": "Biokemi och molekylärbiologianalys med Rust",
          },
          "startTerm": "20231",
          "studyYear": 1,
        },
      ],
    },
  ],
  "status": 200,
}
`;
